src/Core/Services/Implementations/UserService.cs

```diff
@@ -636,33 +637,54 @@ public override Task<IdentityResult> ChangePasswordAsync(User user, string maste
             return IdentityResult.Success;
         }
 
-        public async Task<IdentityResult> SetCryptoAgentKeyAsync(User user, string key, string orgIdentifier)
+        public async Task<IdentityResult> SetKeyConnectorKeyAsync(User user, string key, string orgIdentifier)
         {
             if (user == null)
             {
                 throw new ArgumentNullException(nameof(user));
             }
 
-            if (user.UsesCryptoAgent)
+            if (user.UsesKeyConnector)
             {
-                Logger.LogWarning("Already uses crypto agent.");
+                Logger.LogWarning("Already uses key connector.");
                 return IdentityResult.Failed(_identityErrorDescriber.UserAlreadyHasPassword());
             }
 
             user.RevisionDate = user.AccountRevisionDate = DateTime.UtcNow;
             user.Key = key;
-            user.UsesCryptoAgent = true;
+            user.UsesKeyConnector = true;
 
             await _userRepository.ReplaceAsync(user);
-            // TODO: Use correct event
             await _eventService.LogUserEventAsync(user.Id, EventType.User_ChangedPassword);
 
             await _organizationService.AcceptUserAsync(orgIdentifier, user, this);
 
             return IdentityResult.Success;
         }
 
-        
+        public async Task<IdentityResult> ConvertToKeyConnectorAsync(User user)
+        {
+            if (user == null)
+            {
+                throw new ArgumentNullException(nameof(user));
+            }
+
+            if (user.UsesKeyConnector)
+            {
+                Logger.LogWarning("Already uses key connector.");
+                return IdentityResult.Failed(_identityErrorDescriber.UserAlreadyHasPassword());
+            }
+
+            user.RevisionDate = user.AccountRevisionDate = DateTime.UtcNow;
+            user.MasterPassword = null;
+            user.UsesKeyConnector = true;
+
+            await _userRepository.ReplaceAsync(user);
+            await _eventService.LogUserEventAsync(user.Id, EventType.User_ChangedPassword);
```

Comment:
Is this the right event?

Reply:
Changed to `User_MigratedKeyToKeyConnector`